challenges-services/src/main/java/org/exoplatform/challenges/service/ChallengeServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package org.exoplatform.challenges.service;
+
+import org.exoplatform.challenges.model.Challenge;
+import org.exoplatform.challenges.storage.ChallengeStorage;
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+
+public class ChallengeServiceImpl implements ChallengeService {
+
+  private ChallengeStorage challengeStorage;
+
+  public ChallengeServiceImpl(ChallengeStorage challengeStorage) {
+    this.challengeStorage = challengeStorage;
+  }
+
+  @Override
+  public Challenge createChallenge(Challenge challenge, String username) throws IllegalAccessException {
+    if (challenge == null) {
+      throw new IllegalArgumentException("Challenge is mandatory");
+    }
+    if (challenge.getId() != 0) {
+      throw new IllegalArgumentException("challenge id must be equal to 0");
+    }
+
```

Comment:
We need to check if the user is a manager of the audience space to be able to ceate a challenge inside it

Reply:
done